src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/GenomicAndInsertionPosition.java

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.hellbender.tools.longreads.graph;
+
+import java.util.Objects;
+
+public class GenomicAndInsertionPosition implements Comparable<GenomicAndInsertionPosition> {
+
+    private String contig;
+    private int start;
+    private int insertionOffset;
+
+    public GenomicAndInsertionPosition(final String contig, final int start, final int insertionOffset) {
+        this.contig = contig;
+        this.start = start;
+        this.insertionOffset = insertionOffset;
+    }
+
+    public String getContig() {
+        return contig;
+    }
+
+    public int getStart() {
+        return start;
+    }
+
+    public int getInsertionOffset() {
+        return insertionOffset;
+    }
+
+    /**
+     * Checks for adjacency between two {@link GenomicAndInsertionPosition}s.
+     * Adjacent positions are those that are next to eachother in genomic coordinates, including those with equal
+     * start positions, but insertion offsets that differ by 1.
+     * A {@link GenomicAndInsertionPosition} is NEVER adjacent to itself.
+     * @param that The other {@link GenomicAndInsertionPosition} to check for adjacency.
+     * @return {@code true} iff that {@link GenomicAndInsertionPosition} is adjacent to this {@link GenomicAndInsertionPosition}.
+     */
+    public boolean isAdjacentTo( final GenomicAndInsertionPosition that ) {
+        boolean isAdjacent = false;
+
+        if ( !this.equals(that) ) {
+            if ( getContig().equals(that.getContig()) ) {
+                final int posOffset = Math.abs( this.getStart() - that.getStart() ) +
+                        Math.abs( this.getInsertionOffset() - that.getInsertionOffset() );
+
+                isAdjacent = posOffset == 1;
+            }
+        }
+
+        return isAdjacent;
+    }
+
+    @Override
+    public String toString() {
+        if ( insertionOffset != 0 ) {
+            return String.format("GenomicAndInsertionPosition(%s:%d+%d)", contig, start, insertionOffset);
+        }
+        else {
+            return String.format("GenomicAndInsertionPosition(%s:%d)", contig, start);
+        }
+    }
+
+    @Override
+    public int compareTo(final GenomicAndInsertionPosition o) {
+        int result = this.getContig().compareTo(o.getContig());
+        if ( result == 0 ) {
```

Comment:
There might be a way to rewrite this with `Comparator.andThenComparing`. . .

Reply:
True, but I'm going to hold off on that for now.